src/test/java/org/folio/innreach/repository/MARCTransformationOptionsSettingsRepositoryTest.java

```diff
@@ -0,0 +1,92 @@
+    var MARCTransformOptSetById = repository.getOne(UUID.fromString(PRE_POPULATED_MARC_TRANSFORM_OPT_SET_ID));
+
+    assertNotNull(MARCTransformOptSetById);
+    assertEquals(UUID.fromString(PRE_POPULATED_CENTRAL_SERVER_ID), MARCTransformOptSetById.getCentralServer().getId());
+    assertEquals(UUID.fromString(PRE_POPULATED_CENTRAL_SERVER_RECORD_ID), MARCTransformOptSetById.getCentralServerRecordId());
+    assertTrue(MARCTransformOptSetById.getConfigIsActive());
+    assertEquals(1, MARCTransformOptSetById.getExcludedMARCFields().size());
+    assertEquals(1, MARCTransformOptSetById.getModifiedFieldsForContributedRecords().size());
+  }
+
+  @Test
+  @Sql(scripts = {"classpath:db/central-server/pre-populate-central-server.sql"})
+  void saveMARCTransformOptSet_when_MARCTransformOptSetDoesNotExists() {
+    var MARCTransformOptSet = createMARCTransformOptSet();
+    MARCTransformOptSet.setCentralServer(refCentralServer(UUID.fromString(PRE_POPULATED_CENTRAL_SERVER_ID)));
+    var savedMARCTransformOptSet = repository.save(MARCTransformOptSet);
+
+    assertNotNull(savedMARCTransformOptSet);
+    assertNotNull(savedMARCTransformOptSet.getId());
+    assertEquals(MARCTransformOptSet.getCentralServer(), savedMARCTransformOptSet.getCentralServer());
+    assertEquals(MARCTransformOptSet.getConfigIsActive(), savedMARCTransformOptSet.getConfigIsActive());
+    assertEquals(MARCTransformOptSet.getCentralServerRecordId(), savedMARCTransformOptSet.getCentralServerRecordId());
+    assertEquals(MARCTransformOptSet.getExcludedMARCFields(), savedMARCTransformOptSet.getExcludedMARCFields());
+    assertEquals(MARCTransformOptSet.getModifiedFieldsForContributedRecords(), savedMARCTransformOptSet.getModifiedFieldsForContributedRecords());
+  }
+
+  @Test
+  @Sql(scripts = {"classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/marc-transform-opt-set/pre-populate-marc-transform-opt-set.sql"})
+  void updateMARCTransformOptSet_when_MARCTransformOptSetDataIsValid() {
+    var savedMARCTransformOptSet = repository.getOne(UUID.fromString(PRE_POPULATED_MARC_TRANSFORM_OPT_SET_ID));
+
+    var updatedModifiedFields = savedMARCTransformOptSet.getModifiedFieldsForContributedRecords();
+    updatedModifiedFields.add(createFieldConfig());
+    var updatedConfigIsActive = !savedMARCTransformOptSet.getConfigIsActive();
+    var updatedExcludedMARCFields = savedMARCTransformOptSet.getExcludedMARCFields();
+    updatedExcludedMARCFields.clear();
+    var updatedCentralServerRecordId = UUID.randomUUID();
+
+    savedMARCTransformOptSet.setModifiedFieldsForContributedRecords(updatedModifiedFields);
+    savedMARCTransformOptSet.setConfigIsActive(updatedConfigIsActive);
+    savedMARCTransformOptSet.setExcludedMARCFields(updatedExcludedMARCFields);
+    savedMARCTransformOptSet.setCentralServerRecordId(updatedCentralServerRecordId);
+
+    var updatedMARCTransformOptSet = repository.save(savedMARCTransformOptSet);
```

Comment:
could you please simplify names as it's clear from the context that `MARCTransformationOptionsSettings` is being tested here, no need to specify long names for the variables. This way it is easier to read code: savedMARCTransformOptSet -> found, updatedMARCTransformOptSet -> updated. The same is applicable for other tests cases